### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by colinsurprenant** [logstash-plugins/logstash-output-http#24]
there is a memory leak in the http output. I am currently investigating, it seems related to the Java Thread Executor, the Future handling and the requests callback. Investigating.

Here a simple config to reproduce, given an httpd server running on localhost port 9292

```
input {
    generator {}
}

output {
    stdout {
        codec => dots {}
    }
    http {
        http_method => "get"
        format => "json"
        content_type => "application/json"
        url => "http://localhost:9292/"
    }
}
```

Here's the VisualVM output showing the leak:

<img width="1513" alt="screen shot 2015-10-15 at 2 23 19 pm" src="https://cloud.githubusercontent.com/assets/2010/10523720/e00fb8fe-7349-11e5-965e-e0417d0a3f70.png">

**Comment by colinsurprenant** [comment]
Fix has been merged and plugin version 2.0.5 has been published for logstash v2.x.
A fix backport for logstash 1.5.x will be push shortly. We will not close this issue until the backport is pushed.

**Comment by colinsurprenant** [comment]
fixed in 2.x and 1.5.x in #25 and #26
